File: design/range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums = nums
        

    def sumRange(self, left: int, right: int) -> int:
        mySum = 0
        for num in self.nums[left: right + 1]:
            mySum += num
        return mySum
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: design/range-sum-query-immutable.js
/**
 * @param {number[]} nums
 */
var NumArray = function(nums) {
    this.prefixSum = []
    let total = 0
    for (const num of nums) {
        total += num
        this.prefixSum.push(total)
    }
    
};

/** 
 * @param {number} left 
 * @param {number} right
 * @return {number}

 [-2,0,1,-4,-2,-3]
 */
NumArray.prototype.sumRange = function(left, right) {
    if (left === 0) return this.prefixSum[right]
    return this.prefixSum[right] - this.prefixSum[left - 1]
    
};

/** 
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */